### PR TITLE
[StubIR][Metadata] fix for ObjCBlockVar

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt [hunk 1]
```diff
@@ -560,4 +565,24 @@ private class MappingExtensions(
...
     private val TypeParameterStub.id: Int
         get() = typeParametersInterner.intern(this)
+}
+
+private enum class TypeAliasExpansionMode {
+    /**
+     * Substitution of type alias with fully expanded type.
+     * Typical usage: expanded type of typealias.
+     */
+    ERASE,
```
Comment (SvyatoslavScherbina) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt:575]: Do I understand correctly that this is the only new mode?
`NONE` is equal to `shouldExpandTypeAliases = false`, while `EXPANSION` is equal to `shouldExpandTypeAliases  = true`?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt [hunk 2]
```diff
@@ -451,39 +451,44 @@ private class MappingExtensions(
...
-                km.arguments += typeArguments
-            }
-            if (shouldExpandTypeAliases) {
-                // Abbreviated and expanded types have the same nullability.
-                KmType(flags).also { km ->
-                    km.abbreviatedType = abbreviatedType
-                    val kmUnderlyingType = underlyingType.map(true)
-                    km.arguments += kmUnderlyingType.arguments
-                    km.classifier = kmUnderlyingType.classifier
+    fun StubType.map(mode: TypeAliasExpansionMode = TypeAliasExpansionMode.EXPANSION): KmType = when (this) {
```
Comment (SvyatoslavScherbina) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt:454]: Seems correct from my point of view, but I'm not an expert in metadata.
Have you consulted anyone who is?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt (hunk 1 above)
Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt:575]: Yep.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt (hunk 2 above)
Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt:454]: No, good idea.

**Pull request merged**